File: sdk/java/src/main/java/com/superdev/sdk/exceptions/SuperDevException.java
package com.superdev.sdk.exceptions;

/**
 * Base exception class for all SuperDev SDK errors.
 */
public class SuperDevException extends RuntimeException {

    private final int statusCode;

    /**
     * Creates a new SuperDevException.
     *
     * @param message the error message
     */
    public SuperDevException(String message) {
        super(message);
        this.statusCode = -1;
    }

    /**
     * Creates a new SuperDevException with a cause.
     *
     * @param message the error message
     * @param cause   the underlying cause
     */
    public SuperDevException(String message, Throwable cause) {
        super(message, cause);
        this.statusCode = -1;
    }

    /**
     * Creates a new SuperDevException with a status code.
     *
     * @param message    the error message
     * @param statusCode the HTTP status code
     */
    public SuperDevException(String message, int statusCode) {
        super(message);
        this.statusCode = statusCode;
    }

    /**
     * Returns the HTTP status code associated with this error.
     *
     * @return the status code, or -1 if not applicable
     */
    public int getStatusCode() {
        return statusCode;
    }
}

File: sdk/csharp/src/Exceptions/SuperDevException.cs
using System.Net;

namespace SuperDev.SDK.Exceptions;

/// <summary>
/// Base exception for all SuperDev SDK errors.
/// </summary>
public class SuperDevException : Exception
{
    /// <summary>HTTP status code, or 0 for non-HTTP errors.</summary>
    public HttpStatusCode? StatusCode { get; }

    /// <summary>Structured error details from the API response.</summary>
    public Dictionary<string, object> Details { get; }

    public SuperDevException(string message, HttpStatusCode? statusCode = null, Dictionary<string, object>? details = null)
        : base(message)
    {
        StatusCode = statusCode;
        Details = details ?? new Dictionary<string, object>();
    }

    public SuperDevException(string message, Exception innerException, HttpStatusCode? statusCode = null)
        : base(message, innerException)
    {
        StatusCode = statusCode;
        Details = new Dictionary<string, object>();
    }
}
